RealmSwift/Realm.swift

```diff
@@ -1005,3 +1005,33 @@ extension Realm {
 
 /// The type of a block to run for notification purposes when the data in a Realm is modified.
 public typealias NotificationBlock = (_ notification: Realm.Notification, _ realm: Realm) -> Void
+
+@available(macOS 12.0, tvOS 15.0, iOS 15.0, watchOS 8.0, *)
+extension Realm {
+    private static let asyncOpenQueue = DispatchQueue(label: "io.realm.swift.asyncOpenDispatchQueue")
+    /**
+     Asynchronously open a Realm and deliver it to a block on the given queue.
+
+     Opening a Realm asynchronously will perform all work needed to get the Realm to
+     a usable state (such as running potentially time-consuming migrations) on a
+     background thread before dispatching back to the continiuation. In addition,
+     synchronized Realms wait for all remote content available at the time the
+     operation began to be downloaded and available locally.
+
+     - parameter configuration: A configuration object to use when opening the Realm.
+     - returns: An open Realm.
+     */
+    @discardableResult
```

Comment:
This shouldn't be discardable. The callback-based asyncOpen is because it returns an object that can be used to observe the progress, but using that object isn't required.